modules/git/git_commits/slides/contents.tex

```diff
@@ -0,0 +1,193 @@
+\end{question}
+\pause
+\begin{block}{Two simple rules}
+    \begin{itemize}
+        \item Commit early
+        \item Commit often
+    \end{itemize}
+\end{block}
+\end{frame}
+
+
+\subsection{Commits}
+\begin{frame}
+\begin{block}{General rules for commits}
+   \begin{itemize}
+       \item Single-purpose commits
+       \pause
+       \item Small commits
+       \pause
+       \item Only commit working code
+   \end{itemize}
+\end{block}
+\pause
+\begin{question}
+    How do we know that the code is functional?
+\end{question}
+\end{frame}
+
+\begin{frame}
+\begin{question}
+    How do we know that the code is functional?
+\end{question}
+\pause
+\begin{block}{Solution}
+\begin{itemize}
+    \item We test it!
+    \item Write tests before writing the source code (Test-driven development)
+\end{itemize}
+\end{block}
+\end{frame}
+
+\begin{frame}
+    \begin{alertblock}{Note:}
+    Avoid using \lstinline{git add -A} and \lstinline{git commit -a} as it's easy to lose control over what has been changed in the repository
+    \end{alertblock}
+\end{frame}
+
+\subsection{Messages}
+
+\begin{frame}
+\begin{block}{General rules for messages}
+  \begin{itemize}
+      \item Communicate what changes with this commit along with the intent for it
+      \item Consists of two parts:
+      \begin{enumerate}
+          \item A subject
+          \item A message body
+      \end{enumerate}
+  \end{itemize}
+\end{block}
+\end{frame}
+
+\begin{frame}
+\begin{block}{Subject}
+  \begin{itemize}
+      \item Short (50 characters or less)
+      \item Summarise the commit message
+  \end{itemize}
+\end{block}
+\begin{block}{Style guide}
+  \begin{itemize}
+      \item Imperative mood, present tense
+      \begin{itemize}
+          \item Should be able to finish the sentence "If applied, this commit will..."
+      \end{itemize}
+      \item Not end with a period
+  \end{itemize}
+\end{block}
+\end{frame}
+
+\begin{frame}
+\begin{block}{Message body}
+  \begin{itemize}
+      \item Can be used if more information is necessary
+      \item Should contain detailed answers at least one of the following questions:
+      \begin{itemize}
+          \item What is the motivation/intent of the change?
+          \item How does this new version differ from the old?
+          \item How does this change resolve the issue?
+      \end{itemize}
+  \end{itemize}
+\end{block}
+\begin{block}{Style guide}
+  \begin{itemize}
+    \item Starts with a blank line
+    \item Can be written in bullet form (using either '-' or '*')
+    \begin{itemize}
+        \item Use proper and consistent indentation in that case
+    \end{itemize}
+  \end{itemize}
+\end{block}
+\end{frame}
+
+\section{Summary}
+\begin{frame}
+\begin{block}{Summary}
+\begin{itemize}
+    \item Make small, single-purpose commits of working code as often as possible
+    \item Write messages that clearly communicate what you changed and why
+\end{itemize}
+    
+\end{block}
+    
```

Comment:
Add an exercise or two that they can do, so that they can practice. For instance, suggest a commit message (subject and body) for a change of some tiny code example. Then do it as a discussion rather than a question with a correct answer.

If you have examples of good and bad commit messages that would also be nice, then they can discuss the good and bad aspects of them. And you can cover them in the video as you would do during a tutorial (övning).

Reply:
I was trying to come up with good examples, but my imagination ran dry 🙈. 

Do you have any example messages that I can use? Especially longer ones were both difficult to find in the repos I have access to and difficult to come up with without actually having done a commit.

Reply:
You can probably find some bad examples here: https://github.com/dbosk/introtools/commits/master :rofl:

Reply:
Some of my former colleagues used to have things like "duh", "meh" as their commit messages.